jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletContextHandler.java

```diff
@@ -198,24 +198,33 @@ else if (parent instanceof HandlerCollection)
 
     /**
      * Add EventListener
-     * Adds an EventListener to the list. @see org.eclipse.jetty.server.handler.ContextHandler#addEventListener().
-     * Also adds any listeners that are session related to the SessionHandler.
      *
      * @param listener the listener to add
+     * @return true if the listener was added
+     * @see HttpSessionAttributeListener
+     * @see HttpSessionActivationListener
+     * @see HttpSessionBindingListener
+     * @see HttpSessionListener
+     * @see HttpSessionIdListener
+     * @see ContextHandler#addEventListener(EventListener)
      */
     @Override
-    public void addEventListener(EventListener listener)
+    public boolean addEventListener(EventListener listener)
     {
-        super.addEventListener(listener);
-        if ((listener instanceof HttpSessionActivationListener) ||
-            (listener instanceof HttpSessionAttributeListener) ||
-            (listener instanceof HttpSessionBindingListener) ||
-            (listener instanceof HttpSessionListener) ||
-            (listener instanceof HttpSessionIdListener))
+        if (super.addEventListener(listener))
         {
-            if (_sessionHandler != null)
-                _sessionHandler.addEventListener(listener);
+            if ((listener instanceof HttpSessionActivationListener) ||
+                (listener instanceof HttpSessionAttributeListener) ||
+                (listener instanceof HttpSessionBindingListener) ||
+                (listener instanceof HttpSessionListener) ||
+                (listener instanceof HttpSessionIdListener))
+            {
+                if (_sessionHandler != null)
```

Comment:
Should you return the boolean from SessionHandler.addEventListener()?

Reply:
No, because the whole branch here is because the `super.addEventListener` returned true, thus the listener has been added.